src/protocols/rdp/channels/cliprdr.c

```diff
@@ -283,7 +283,8 @@ static UINT guac_rdp_cliprdr_format_list(CliprdrClientContext* cliprdr,
     if (guac_rdp_cliprdr_format_supported(format_list, CF_TEXT))
         return guac_rdp_cliprdr_send_format_data_request(cliprdr, CF_TEXT);
 
-    /* Ignore any unsupported data */
+    /* Suppose that any unsupported data is raw data (in this case, CF_RAW) */
+    guac_rdp_cliprdr_send_format_data_request(cliprdr, CF_RAW);
```

Comment:
Why?

Reply:
guac_rdp_cliprdr_format_list() function is called when copying any data in the remote server.
In this function, we need to tell the remote server whether the remote client supports the clipboard format of the remote clipboard.
Users can copy any kind of data including text in the remote server.
Because I thought that CF_RAW indicates any kind of clipboard data, I selected the clipboard format as CF_RAW.

Reply:
If the format of the data in the remote clipboard is not supported (is not handled by one of the `if` statements above), why are we requesting that the RDP server send us that data?

Reply:
> why are we requesting that the RDP server send us that data?
Because we need to reset the local clipboard. That's, to get the opportunity to reset the local clipboard when users copy any data other than text.

The added new code of `guac_rdp_cliprdr_format_list()` function let the remote server to call `guac_rdp_cliprdr_format_data_response()` function according to **"Remote Desktop Protocol: Clipboard Virtual Channel Extension"**.
If users copy any kind of data other than text, so we can use `guac_rdp_cliprdr_format_data_response()` function to reset the local clipboard.

Reply:
Yes, but why reset the local clipboard? Why is resetting the local clipboard the solution to not erasing what's on the remote clipboard?

Reply:
In the Apache Guacamole, the process related to the clipboard has one cyclic loop.
If the remote clipboard is changed, the content of the remote clipboard is reflected in the local clipboard.
This produces one regression to reflect the local clipboard to the remote clipboard, again.

So, I thought that we have to reset the local clipboard when copying any data other than text in the remote server.
Then, we can check the length of the local clipboard in `guac_rdp_clipboard_end_handler()` function to decide whether we maintain the clipboard content of the remote server.

Reply:
Also, you can see the following image for the workflow when copying any data in the remote server:

![image](https://user-images.githubusercontent.com/52309150/102897151-3750cc00-4470-11eb-940b-b4b343ef2002.png)